worldedit-core/src/main/java/com/sk89q/worldedit/extension/factory/parser/DefaultBlockParser.java

```diff
@@ -272,21 +280,41 @@ private BaseBlock parseLogic(String input, ParserContext context) throws InputPa
 
         if (state == null) {
             String typeString;
-            String stateString = null;
+
             int stateStart = blockAndExtraData[0].indexOf('[');
-            if (stateStart == -1) {
+            int nbtStart = blockAndExtraData[0].indexOf('{');
+            int typeEnd = stateStart == -1 ? nbtStart : stateStart;
+
+            if (typeEnd == -1) {
                 typeString = blockAndExtraData[0];
             } else {
-                typeString = blockAndExtraData[0].substring(0, stateStart);
+                typeString = blockAndExtraData[0].substring(0, typeEnd);
+            }
+
+            String stateString = null;
+            if (stateStart != -1) {
                 if (stateStart + 1 >= blockAndExtraData[0].length()) {
                     throw new InputParseException(TranslatableComponent.of("worldedit.error.parser.hanging-lbracket", TextComponent.of(stateStart)));
                 }
                 int stateEnd = blockAndExtraData[0].lastIndexOf(']');
                 if (stateEnd < 0) {
                     throw new InputParseException(TranslatableComponent.of("worldedit.error.parser.missing-rbracket"));
                 }
-                stateString = blockAndExtraData[0].substring(stateStart + 1, blockAndExtraData[0].length() - 1);
+                stateString = blockAndExtraData[0].substring(stateStart + 1, stateEnd);
             }
+
+            String nbtString = null;
+            if (nbtStart != -1) {
+                if (nbtStart + 1 >= blockAndExtraData[0].length()) {
+                    throw new InputParseException(TranslatableComponent.of("worldedit.error.parser.hanging-lbrace", TextComponent.of(nbtStart)));
+                }
+                int nbtEnd = blockAndExtraData[0].lastIndexOf('}');
```

Comment:
Similarly here, we might cut off NBT early if there's `}` in an NBT string, and someone doesn't put one at the end, resulting in a weird error.

Reply:
This will result in an invalid NBT error